docs/stdlib/select.md

```diff
@@ -66,3 +66,35 @@ Options      |         Description           | Required?
 
 Only the point with the exact median value for the `field` name
 provided will be forwarded.
+
+## top - sub
+
+```
+... | select.top -n number -by field | ...
+```
+
+Options  |                              Description                              | Required?
+-------- | --------------------------------------------------------------------- | --------- :
+`n`      | how many of the fields with the highest values of `by` to keep        | Yes
+`by`     | name of the field to sort by before figuring out the top `n` to keep  | Yes
+
+Let the `n` points that have the highest value of `by` pass through. 
+
+**Note** since we use sort to sort by the `by` field we end up not emitting
```

Comment:
@dmehra Revisiting the decision to drop time in `sort` in non-batched streams means either:

  1. Revisiting the requirement for time to be non-decreasing in points.
  2. Changing the semantics of `sort` to take times from points, sort them, and assign them back one by one to points sorted by specified criteria.

I’m not opposed to either, it just needs thinking through. Obviously, 2 is more localized change, but the association between points and their times would be lost (which is not intuitive and may prove hurt some use cases). It also wouldn’t make sense to do it if we want to do 1 at tome point. As for 1, Henri wanted to do it for quite some time, we just need to think through all the consequences.

Probably worth filing an issue. Also, summoning @bkutil.